alphalens/utils.py

```diff
@@ -852,12 +893,26 @@ def std_conversion(period_std, base_period):
     return period_std / np.sqrt(conversion_factor)
 
 
-def get_forward_returns_columns(columns):
+def get_forward_returns_columns(columns, require_exact_day_multiple=False):
     """
     Utility that detects and returns the columns that are forward returns
     """
-    pattern = re.compile(r"^(\d+([Dhms]|ms|us|ns))+$", re.IGNORECASE)
-    valid_columns = [(pattern.match(col) is not None) for col in columns]
+
+    # If exact day multiples are required in the forward return periods,
+    # drop all other columns (e.g. drop 3D12h).
+    if require_exact_day_multiple:
```

Comment:
This case probably needs a test; I can add.

Reply:
ty!